lib/logflare_web/live/search_live/logs_search_lv.ex

```diff
@@ -837,17 +897,31 @@ defmodule LogflareWeb.Source.SearchLV do
          %{suggested_keys: suggested_keys},
          %{assigns: %{force_query: false}} = socket
        ) do
-    suggested_present =
+    {required, suggested} =
       suggested_keys
       |> String.split(",")
       |> Enum.map(fn
         "m." <> suggested_field -> "metadata." <> suggested_field
         suggested_field -> suggested_field
       end)
-      |> Enum.all?(fn suggested_field ->
+      |> Enum.split_with(fn suggested_field -> String.ends_with?(suggested_field, "!") end)
+
+    suggested_present =
+      Enum.all?(suggested, fn suggested_field ->
         Enum.find(lql_rules, fn %{path: path} -> path == suggested_field end)
       end)
 
-    if suggested_present, do: {:ok, socket}, else: {:error, :suggested_field_not_found}
+    required_present =
+      Enum.all?(required, fn required_field ->
+        Enum.find(lql_rules, fn %{path: path} ->
+          path == String.trim_trailing(required_field, "!")
```

Comment:
Trim once instead of per rule:

```suggestion
        required_field = String.trim_trailing(required_field, "!")

        Enum.find(lql_rules, fn %{path: path} ->
          path == required_field
```